### PR TITLE
[Woo POS] Align Products and Cart toolbars sizes

### LINKED ISSUE #12394: [Woo POS] Products and Cart toolbars' shadows not aligned
Products and Cart toolbars sizes (shadows) are not aligned:

Image

### PR DESCRIPTION
**Pull request by samiuelson**
Don't merge until the base branch is merged to trunk.
Closes: #12394 

### Description

This PR aligns the sizes of the Products and Cart toolbars so that the shadows visible after both columns are scrolled are at the same level.

### Steps to reproduce

1. Add many items to cart and scroll both the Products and Cart columns 
2. Verify that shadows at the top of each column are not aligned

### Testing information

The shadows at the top of each column (product and cart lists) should be at the same level.

### The tests that have been performed

I verified that both shadows are at the correct position.

### Images/gif

|beofre|after|
|---|---|
|Screenshot_20240822_085531_WooCommerce (Dev)|Screenshot 2024-08-22 at 16 48 57|

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/232">20.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/component/WooPosLazyColumn.kt [hunk 1]
```diff
@@ -48,13 +49,13 @@ fun WooPosLazyColumn(
         }
 
         if (showShadow.value) {
-            Card(
+            Surface(
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/component/WooPosLazyColumn.kt:52]: Not related; minor layout hierarchy clean up

**Review by backwardstruck — COMMENTED** [review]
Screenshot_20240822_145710

**Review by backwardstruck — APPROVED** [review]
Good catch.

**Pull request merged**